enumerable.rb

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env ruby
+
+# Some methods will be added to this module
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # definition of the my_each method
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self do # rubocop:disable Style/For
+      yield(element)
+    end
+    self
+  end
+
+  # definition of the my_each_with_index method
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    my_each do |element|
+      yield(element, index)
+      index += 1
+    end
+    self
+  end
+
+  # definition of the my_select method
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |element|
+      new_array << element if yield(element)
+    end
+    new_array
+  end
+
+  # definition of the my_all? method
+
+  def my_all?(pattern = nil) # rubocop:disable CyclomaticComplexity, PerceivedComplexity
+    return true unless block_given? || !pattern.nil?
+
+    my_each do |element|
+      if block_given?
+        return false unless yield(element)
+      elsif pattern.is_a?(Class) || pattern.is_a?(Regexp)
+        return false unless pattern === element # rubocop:disable Style/CaseEquality
+      else
+        return false unless element == pattern
```

Comment:
You seem to add some breaking changes after my review. You handled the cases @Forison is talking about properly before.

Please check the ruby doc and use it as guard.

You still need to handle cases where no argument is given, that is `pattern` is `nil`.

- Kindly remove this line `return true unless block_given? || !pattern.nil?` since we want to handle cases where block is not given and cases where pattern is given or not
- Take good advantage of `===`, case equality can handle all case where an argument is given. So use it for when `pattern` is not `nil`
- Finally handle case where no argument is given. I think this part was implemented before. Loop through the element and check if they are truthy or falsy values (as the case may be) and return the necessary Boolean.

### Do this for other two similar methods (my_any and my_none)

Again, test your methods with the examples in the ruby doc, see link in previous reviews

Chat me up on slack or comment here if you need further clarification.

#### This is not a CRR